lighthouse-core/gather/driver.js

```diff
@@ -630,11 +630,11 @@ class Driver {
      * @param {() => void} resolve
      */
     function checkForQuiet(driver, resolve) {
```

Comment:
`_waitForCPUIdle` is definitely confusing :)

It would be nice to simplify things a little bit while in here. There are a few ways to do that, but one simple way (accomplishing the same thing you're doing here), just switching to `async` and flattening the `evaluateAsync` call:
```js
/**
 * @param {Driver} driver
 * @param {() => void} resolve
 * @return {Promise<void>}
 */
async function checkForQuiet(driver, resolve) {
  if (cancelled) return;
  const timeSinceLongTask = await driver.evaluateAsync(checkForQuietExpression);
  if (cancelled) return;

  if (typeof timeSinceLongTask === 'number') {
    if (timeSinceLongTask >= waitForCPUQuiet) {
      log.verbose('Driver', `CPU has been idle for ${timeSinceLongTask} ms`);
      resolve();
    } else {
      log.verbose('Driver', `CPU has been idle for ${timeSinceLongTask} ms`);
      const timeToWait = waitForCPUQuiet - timeSinceLongTask;
      lastTimeout = setTimeout(() => checkForQuiet(driver, resolve), timeToWait);
    }
  }
}
```